includes/RestApi/FlowController.php

```diff
@@ -100,31 +97,32 @@ public function save_onboarding_flow_data( \WP_REST_Request $request ) {
 				array( 'status' => 404 )
 			);
 		}
-
-		if ( ! ( $flow_data = $this->read_details_from_wp_options() ) ) {
-			  $flow_data              = Flows::get_data();
-			  $flow_data['createdAt'] = time();
-			  // update default data if flow type is ecommerce
-			  $flow_data = $this->update_default_data_for_ecommerce( $flow_data );
-			  $this->save_details_to_wp_options( $flow_data );
+		
+		$flow_data = FlowService::read_flow_data_from_wp_option();
+		$updated_flow_data = array();
+		$flow_data = FlowService::update_post_call_data_recursive( $flow_data, $updated_flow_data, $params );
+		if(!is_array($flow_data)) {
+			return new \WP_Error(
+				'wrong_param_type_provided',
+				"Wrong Parameter Type Provided : ". $flow_data,
+				array( 'status' => 404 )
+			);
 		}
 
-		foreach ( $params as $key => $param ) {
-			if ( $value = $this->array_search_key( $key, $flow_data ) === false ) {
-				return new \WP_Error(
-					'wrong_param_provided',
-					"Wrong Parameter Provided : $key",
-					array( 'status' => 404 )
-				);
-			}
+		$default_flow_data = FlowService::get_default_flow_data();
+		$mismatch_key = FlowService::check_key_in_nested_array($params, $default_flow_data);
+		if ( ! empty($mismatch_key) )  {
+			return new \WP_Error(
+				'wrong_param_provided',
+				"Wrong Parameter Provided",
+				array( 'status' => 404 , 'Mismatched Parameter(s)' => $mismatch_key)
+			);
 		}
 
-		$flow_data = array_replace_recursive( $flow_data, $params );
-
 		// update timestamp once data is updated
-		$flow_data['updatedAt'] = time();
+		$flow_data['updatedAt'] = strval(time());
```

Comment:
Can we add a comment as to why this is made a string? And also should we updated `createdAt` to be a string as well?

Reply:
I meant add a comment in the code as to why this is made a string.

Reply:
As discussed, this change was reverted and the blueprint datatype for the same was made an integer